superset/explore/utils.py

```diff
@@ -24,15 +24,24 @@
     ChartNotFoundError,
 )
 from superset.charts.dao import ChartDAO
+from superset.commands.exceptions import DatasourceNotFoundValidationError
 from superset.datasets.commands.exceptions import (
     DatasetAccessDeniedError,
     DatasetNotFoundError,
 )
 from superset.datasets.dao import DatasetDAO
+from superset.utils.core import DatasourceType
 from superset.views.base import is_user_admin
 from superset.views.utils import is_owner
 
 
+def check_datasource_access(datasource_id: int, datasource_type: str) -> Optional[bool]:
+    if datasource_id:
+        if datasource_type == DatasourceType.TABLE:
```

Comment:
I'm not sure if `check_dataset_access` depends on the `datasource_type`. Shouldn’t the check occur independently of the data source type?

Reply:
Right now, it's not dependent because there's only one type, but when we start adding other types, we'll do something like this: 
```
def check_query_access(query_id: int) -> Optional[bool]:
    if query_id:
        query = QueryDAO.find_by_id(query_id)
        if query:
            can_access_datasource = security_manager.raise_for_access(
                query=query)
            if can_access_datasource:
                return True
    raise QueryNotFoundError
```

Reply:
I'll leave a comment to make that more clear.